vendor/openfx/OCIOColorSpace.cpp

```diff
@@ -0,0 +1,224 @@
+                        os.str());
+        }
+    }
+
+    else if (paramName == "swap_src_dst")
+    {
+        int srcCsIdx, dstCsIdx;
+        srcCsNameParam_->getValue(srcCsIdx);
+        dstCsNameParam_->getValue(dstCsIdx);
+
+        // Swap src and dst color space indices
+        srcCsNameParam_->setValue(dstCsIdx);
+        dstCsNameParam_->setValue(srcCsIdx);
+    }
+    else
+    {
+        contextParamChanged(*this, paramName);
+    }
+}
+
+void OCIOColorSpaceFactory::describe(OFX::ImageEffectDescriptor& desc)
+{
+    // Labels
+    desc.setLabels("OCIOColorSpace", "OCIOColorSpace", "OCIOColorSpace");
+    desc.setPluginGrouping("OpenColorIO");
+
+    // Supported contexts
+    desc.addSupportedContext(OFX::eContextFilter);
+    desc.addSupportedContext(OFX::eContextGeneral);
+
+    // Supported pixel depths
+    desc.addSupportedBitDepth(OFX::eBitDepthUByte);
+    desc.addSupportedBitDepth(OFX::eBitDepthUShort);
+    desc.addSupportedBitDepth(OFX::eBitDepthHalf);
+    desc.addSupportedBitDepth(OFX::eBitDepthFloat);
+
+    // Flags
+    desc.setRenderTwiceAlways(false);
+}
+
+void OCIOColorSpaceFactory::describeInContext(OFX::ImageEffectDescriptor& desc, 
+                                              OFX::ContextEnum /*context*/)
+{
+    // Create the mandated source clip
+    OFX::ClipDescriptor * srcClip = desc.defineClip(
+        kOfxImageEffectSimpleSourceClipName);
+
+    srcClip->addSupportedComponent(OFX::ePixelComponentRGBA);
+    srcClip->addSupportedComponent(OFX::ePixelComponentRGB);
+
+    // Create the mandated output clip
+    OFX::ClipDescriptor * dstClip = desc.defineClip(kOfxImageEffectOutputClipName);
+
+    dstClip->addSupportedComponent(OFX::ePixelComponentRGBA);
+    dstClip->addSupportedComponent(OFX::ePixelComponentRGB);
+
+    // Define parameters
+    OFX::PageParamDescriptor * page = desc.definePageParam("Controls");
+
+    // Src color space
+    OFX::ChoiceParamDescriptor * srcCsNameParam = defineCsNameParam(
+        desc, 
+        "src_cs", 
+        "Src Color Space", 
+        "Source color space name", 
+        0);
+    page->addChild(*srcCsNameParam);
+
+    // Dst color space
+    OFX::ChoiceParamDescriptor * dstCsNameParam = defineCsNameParam(
+        desc, 
+        "dst_cs", 
+        "Dst Color Space", 
+        "Destination color space name", 
+        0);
+    page->addChild(*dstCsNameParam);
+
+    // Inverse
+    OFX::BooleanParamDescriptor * inverseParam = defineBooleanParam(
+        desc, 
+        "inverse", 
+        "Inverse", 
+        "Invert the transform",
+        0);
+    page->addChild(*inverseParam);
+
+    // Swap color spaces
+    OFX::PushButtonParamDescriptor * swapSrcDstParam = definePushButtonParam(
+        desc, 
+        "swap_src_dst", 
+        "Swap color spaces", 
+        "Sap src and dst color spaces",
+        0);
+    page->addChild(*swapSrcDstParam);
+
+    // Context overrides
+    defineContextParams(desc, page);
```

Comment:
The reason the context values are blank by default is that they will only override the context if they are non-empty, otherwise the default OCIO-evaluated value is used. Nuke uses a checkbox to toggle overrides, but I was attempting to simplify it here. If I were to set the value with the current context variable value by default, it would bake the current env state into the plugin, which may not be desired, since it would be persistent.

As for the labels not appearing, I'm curious if Flame is failing to display the labels because they are within a (collapsible) group parameter. Perhaps an issue in the host implementation? The labels are set the same as for all other parameters and appear in Resolve and Natron. That's the only clear difference I can see in terms of the code.